src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -201,30 +199,64 @@ private string GetBundleFlavorForCurrentEnvironment()
 
         private async Task<bool> TryDownloadZipFileAsync(Uri zipUri, string filePath, HttpClient httpClient)
         {
-            _logger.DownloadingZip(zipUri, filePath);
-            var response = await httpClient.GetAsync(zipUri);
-            if (!response.IsSuccessStatusCode)
+            async Task AttemptAsync()
             {
-                _logger.ErrorDownloadingZip(zipUri, response);
-                return false;
+                _logger.DownloadingZip(zipUri, filePath);
+                var response = await httpClient.GetAsync(zipUri);
+                if (!response.IsSuccessStatusCode)
+                {
+                    _logger.ErrorDownloadingZip(zipUri, response);
+                    throw new InvalidOperationException($"StatusCode {(int)response.StatusCode} ({response.StatusCode})");
+                }
+
+                using (var content = await response.Content.ReadAsStreamAsync())
+                using (var stream = new FileStream(filePath, FileMode.Create, FileAccess.Write, FileShare.None, bufferSize: 4096, useAsync: true))
+                {
+                    await content.CopyToAsync(stream);
+                }
+
+                _logger.DownloadComplete(zipUri, filePath);
             }
 
-            using (var content = await response.Content.ReadAsStreamAsync())
-            using (var stream = new FileStream(filePath, FileMode.Create, FileAccess.Write, FileShare.None, bufferSize: 4096, useAsync: true))
+            try
             {
-                await content.CopyToAsync(stream);
+                await Utility.InvokeWithRetriesBackoffAsync(
+                    AttemptAsync,
+                    maxRetries: 5,
+                    logger: _logger,
+                    unit: TimeSpan.FromMilliseconds(500),
+                    min: TimeSpan.FromMilliseconds(200),
+                    max: TimeSpan.FromSeconds(5));
+                return true;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Failed to download extension bundle after retries. Uri={zipUri} FilePath={filePath} Disk={GetDiskUsageSafe(filePath)}");
+                return false;
             }
-
-            _logger.DownloadComplete(zipUri, filePath);
-            return true;
         }
 
-        private async Task<string> GetLatestMatchingBundleVersionAsync()
+        private string GetDiskUsageSafe(string path)
```

Comment:
Rather than returning "n/a", would returning the actual failure reason make troubleshooting easier?